src/Database/DisposableFullTextSearch.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2024, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Database;
+
+/**
+ * Full-text search on a haystack string that isn't present in the database.
+ * The haystack is inserted in a temporary table with a FULLTEXT index, then any number of
+ * matches can be performed on it before the row is deleted when the class instance is destroyed,
+ * either manually or at the end of the script at the latest.
+ */
+class DisposableFullTextSearch
+{
+	private Database $db;
+	/** @var int Unique identifier of the haystack in the database. */
+	private int $identifier;
+
+	public function __construct(Database $database, string $haystack)
+	{
+		$this->db = $database;
+
+		do {
+			// Unique identifier generation. Two DisposableFullTextSearch object should never have the same as the first object destruction
+			// would delete both check-full-text-search rows before the second object destruction is called, leading to unexpected behavior.
+			// Maximum value is indicated by the INT UNSIGNED type of the check-full-text-search.pid field
+			$this->identifier = random_int(0, pow(2, 32) - 1);
+		} while($this->db->insert('check-full-text-search', ['pid' => $this->identifier, 'searchtext' => $haystack]));
```

Comment:
Can you add some limit, so that there isn't an endless loop when the database becomes unavailable? I guess that for example `10` should be a good number, since it is extremely unlikely that you would had 10 hits. Otherwise of course you could have a quick check via `exists`.

Reply:
`exists()` will also return `false` if the database isn't available, so there's no benefit to use a second query.

Reply:
When `exists` returned `false`, then the system would perform some `insert` that would fail as well. But this wouldn't lead to some endless loop like it would now. Especially when this was called via a worker, then in the worst case that loop would continue for several hours.

Reply:
Got it. What about the behavior when we can't insert the search text row? Are we fine with silently failing any subsequent match calls or should we throw an exception?

Reply:
I'm fine with silently failing, since the only condition where this should happen is some database issue. And such a situation will cause a lot of problems at many places.